packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -1650,6 +1650,14 @@ class DropdownButtonFormField<T> extends FormField<T> {
            final InputDecoration effectiveDecoration = decorationArg.applyDefaults(
              Theme.of(field.context).inputDecorationTheme,
            );
+
+           final bool showSelectedItem = !(items == null || items.where((DropdownMenuItem<T> item) => item.value == state.value).isEmpty);
+           final bool showHintOrDisabledHint =
+             (onChanged != null)
+             ? hint != null && !showSelectedItem
+             : (hint != null || disabledHint != null) && !showSelectedItem;
```

Comment:
We should also consider the `disabledHint` shown when the menu is disabled.

Reply:
Good catch!